src/UriInterface.hh

```diff
@@ -173,9 +173,21 @@ interface UriInterface {
    *
    * @see https://tools.ietf.org/html/rfc3986#section-2
    * @see https://tools.ietf.org/html/rfc3986#section-3.4
+   * @return string The percent-encoded query string
    */
   public function getQuery(): string;
 
+  /**
+   * Retrieve the query params of the URI.
+   *
+   * If no query params dict is present, this method MUST return an empty dict.
+   *
+   * The value returned MUST be the same as the value passed to withRawQuery()
+   *
+   * @return dict<string, ?string> The query params dict with unencoded (raw) values
+   */
+  public function getRawQuery(): dict<string, ?string>;
```

Comment:
I think this is the wrong way around: I would expect `getRawQuery()` to involve less processing than `getQuery()`

Reply:
Could you tell me a bit more about what do you expect to see here?

In my opinion, the implementation of `getRawQuery()` just needs to return the dictionary that was passed to `withRawQuery()` so there is no processing needed.